crates/pikelet-concrete/src/elaborate/mod.rs

```diff
@@ -639,44 +616,12 @@ pub fn infer_term(
         // I-CASE
         raw::Term::Case(span, ref raw_head, ref raw_clauses) => {
             let (head, head_ty) = infer_term(context, raw_head)?;
-            let mut ty = None;
-
-            // TODO: ensure that patterns are exhaustive
-            let clauses = raw_clauses
-                .iter()
-                .map(|raw_clause| {
-                    let (raw_pattern, raw_body) = raw_clause.clone().unbind();
-                    let (pattern, declarations) = check_pattern(context, &raw_pattern, &head_ty)?;
-
-                    let (body, body_ty) = {
-                        let mut body_context = context.clone();
-                        for (free_var, ty) in declarations {
-                            body_context.insert_declaration(free_var, ty);
-                        }
-                        infer_term(&body_context, &raw_body)?
-                    };
-
-                    match ty {
-                        None => ty = Some(body_ty),
-                        // FIXME: use common subtype?
-                        Some(ref ty) if RcValue::term_eq(&body_ty, ty) => {},
-                        Some(ref ty) => {
-                            return Err(TypeError::Mismatch {
-                                span: raw_body.span(),
-                                found: Box::new(context.resugar(&body_ty)),
-                                expected: Box::new(context.resugar(ty)),
-                            });
-                        },
-                    }
-
-                    Ok(Scope::new(pattern, body))
-                })
-                .collect::<Result<_, TypeError>>()?;
 
-            match ty {
-                Some(ty) => Ok((RcTerm::from(Term::Case(head, clauses)), ty)),
-                None => Err(TypeError::AmbiguousEmptyCase { span }),
+            Err(InternalError::Unimplemented {
+                span: Some(span),
+                message: "infer raw::Term::Case".to_owned(),
             }
+            .into())
```

Comment:
Struggling to figure out how to structure this pattern match compilation... I have to take a list of clauses and compile them into the core language... 🤔